plugins/web-terminal/src/components/NamespacePickerDialog/NamespacePickerDialog.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  FormControl,
+  Select,
+  MenuItem,
+  makeStyles,
+  Button,
+} from '@material-ui/core';
+import React, { useEffect } from 'react';
+
+const useStyles = makeStyles({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    justifyContent: 'center',
+    marginBottom: '10px',
+  },
+  buttonmargin: {
+    marginTop: '10px',
+    marginBottom: '10px',
+  },
+});
+
+type props = {
+  onInit: () => Promise<any>;
+  open: boolean;
+  previousNamespace: string;
+  handleClose: () => void;
+  onSubmit: (selectedNamesapce: string) => void;
+};
+
+export const NamespacePickerDialog = ({
+  onInit,
+  open,
+  previousNamespace,
+  handleClose,
+  onSubmit,
+}: props) => {
+  const classes = useStyles();
+  const namespaceRef = React.useRef<HTMLInputElement>(null);
+  const [selectedNamespace, setSelectedNamespace] = React.useState('');
+  const [namespaces, setNamespaces] = React.useState([]);
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setSelectedNamespace(event.target.value as string);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onSubmit(selectedNamespace);
+  };
+
+  useEffect(() => {
+    onInit().then(setNamespaces);
+  }, [onInit]);
+
+  return (
+    <Dialog open={open} onClose={handleClose}>
+      <DialogTitle>Choose a namespace</DialogTitle>
+      <DialogContent>
+        <DialogContentText>
+          Unable to create devworkspace resource in {previousNamespace}{' '}
+          namespace, please choose a different one
+        </DialogContentText>
+        <form onSubmit={handleSubmit} className={classes.container}>
+          <FormControl>
+            <Select
+              name="namespace"
+              inputRef={namespaceRef}
+              onChange={handleChange}
+              value={selectedNamespace}
+              displayEmpty
+              required
+            >
+              <MenuItem value="" disabled>
+                Select a namespace
+              </MenuItem>
+              {namespaces.map(namespace => {
+                return (
+                  <MenuItem key={namespace} value={namespace}>
+                    {namespace}
+                  </MenuItem>
+                );
+              })}
+            </Select>
+            <Button
+              type="submit"
+              color="primary"
+              variant="contained"
+              className={classes.buttonmargin}
+            >
+              Submit
+            </Button>
+          </FormControl>
+        </form>
```

Comment:
I think we should show a loading indicator while namespaces are being loaded. This way the dropdown remains empty without any user feedback until loaded...

Reply:
We can do this in next step/iteration if you want.

Reply:
We can try to do it this iteration :) wdyt about [this solution](https://github.com/operate-first/service-catalog/pull/214/files#diff-a740942b45b619844d609cc7e4844a813311ca726145009fbe17737ff2859057R68)
Resulting modal when loading namespaces
![image](https://user-images.githubusercontent.com/47832967/219721183-fba1ffc5-5fd1-4603-a884-865116c6c965.png)

Reply:
This looks great! :100: